server/pages/api/downloadPass.ts

```
@@ -65,20 +66,37 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
                                 console.log('then timestamp:', timestamp)
 
                                 // Lookup ENS name
-                                // TODO
-
-                                // Populate the pass template
-                                const filePath : string = Passes.downloadPass(Platform.Apple, passportID, timestamp, address)
-                                console.log('filePath:', filePath)
-
-                                // Serve the pass download to the user
-                                const fileName = `passport_${address}.pkpass`
-                                console.log('fileName:', fileName)
-                                res.setHeader('Content-Disposition', `attachment;filename=${fileName}`)
-                                res.setHeader('Content-Type', 'application/vnd.apple.pkpass')
-                                res.setHeader('Content-Length', fs.statSync(filePath).size)
-                                const readStream = fs.createReadStream(filePath)
-                                readStream.pipe(res)
+                                let ensName : string = ''
+                                lookupEnsName(address)
+                                        .then((result: any) => {
+                                            console.log('then result:', result)
+                                            if (result != null) {
+                                                ensName = result
+                                            } else {
+                                                console.warn('ENS name not found for address')
+                                            }
+                                        })
+                                        .catch((error) => {
+                                            console.error('catch error:', error)
```

This is unusual but maybe it is fine, do we have anything that'll help us remember that when fetching ENS name, empty is fine, and error state is treated as empty, does it matter?

---

@johnmark13 Are you saying the whole `catch {}` block can be removed?

---

No, if you remove the catch, the whole chunk of code will fail when ENS has as error.

 I just wondered what can cause the error, and if there is an error there should we fail rather than just logging and continuing. 

I guess that depends on how important the ENS name is, and if it can get updated at a later date.

---

@johnmark13 Getting the ENS name is not crucial, because we use the regular ETH address as the fallback if something fails with the ENS lookup. That's why the `catch {}` block is not doing anything besides logging the error.

---

> I guess that depends on how important the ENS name is, and if it can get updated at a later date.

@johnmark13 Yes, the user can call the download URL again, and download a new `.pkpass` file.  We can also distribute pass updates: #9